File: src/main/java/dev/codex/armatureskin/packageformat/Mc3dSkinManifest.java
package dev.codex.armatureskin.packageformat;

import java.util.List;

public record Mc3dSkinManifest(
        String format,
        int formatVersion,
        String packageId,
        String packageVersion,
        String displayName,
        String author,
        String model,
        List<String> textures,
        String preferredTexture
) {
    public List<String> textures() {
        return textures == null ? List.of() : List.copyOf(textures);
    }
}

File: unity/Minecraft3DSkinExporter/Editor/Mc3dSkinManifest.cs
using System;
using System.Collections.Generic;

namespace Sakutarooo.Minecraft3DSkinExporter.Editor
{
    [Serializable]
    public sealed class Mc3dSkinManifest
    {
        public string format = "mc3dskin";
        public int formatVersion = 1;
        public string packageId;
        public string packageVersion = "1.0.0";
        public string displayName;
        public string author;
        public string model;
        public string defaultTexture;
        public List<string> textures = new();
        public string rig = "generic";
        public string target = "player_skin";
        public string contentSha256 = "";
    }
}
